Lombiq.Hosting.Tenants.Management/Controllers/ShellSettingsEditorController.cs

```diff
@@ -47,29 +49,56 @@ public async Task<IActionResult> Edit(ShellSettingsEditorViewModel model)
             return NotFound();
         }
 
-        var settingsDictionary = JsonConvert.DeserializeObject<Dictionary<string, string>>(model.Json);
-        var newSettings = new Dictionary<string, string>();
-
-        foreach (var key in settingsDictionary.Keys.Where(key => string.IsNullOrEmpty(settingsDictionary[key])))
+        model.Json ??= "{}";
+        if (!IsValidJson(model.Json))
         {
-            settingsDictionary[key] = null;
+            return RedirectToAction(
+                nameof(AdminController.Edit),
+                typeof(AdminController).ControllerName(),
+                new
+                {
+                    area = "OrchardCore.Tenants",
+                    id = model.TenantId,
+                });
         }
 
-        foreach (var key in settingsDictionary.Keys)
+        var settingsDictionary = new JsonConfigurationParser().ParseConfiguration(model.Json);
```

Comment:
This should never return `null` (what it seems to do) so the below null checks on `settingsDictionary` are not necessary.